lurk-verify/src/main.rs

```diff
@@ -0,0 +1,46 @@
+use clap::Parser;
+use clap_verbosity_flag::{Verbosity, WarnLevel};
+use log::{error, info};
+use lurk::proof::nova::public_params;
+use std::io;
+use std::path::PathBuf;
+
+use lurk_verify::{LurkProof, VerifyError, S1};
+
+#[derive(Parser, Debug)]
+#[clap(version, about, long_about = None)]
+struct Cli {
+    /// Be verbose
+    #[clap(flatten)]
+    verbose: Verbosity<WarnLevel>,
+
+    /// Path to proof to be verified
+    #[clap(short, long)]
+    proof: PathBuf,
+}
+
+impl Cli {
+    fn verify(&self) -> Result<(), VerifyError> {
+        let proof = LurkProof::<S1>::read_from_path(&self.proof)?;
+        let pp = public_params(proof.reduction_count.count());
+        let result = proof.verify(&pp)?;
+
+        serde_json::to_writer(io::stdout(), &result)?;
+
+        if result.verified {
+            info!("Verification succeeded");
+        } else {
+            error!("Verification failed");
```

Comment:
I think maybe failed verification should exit with non-zero exit code. That would simplify plausible use cases of this as a CLI tool.